buildconfig/Jenkins/Conda/conda-buildscript-linux

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash -ex
+
+# This script expects to be in a POSIX environment, it will run our CI workflow on a POSIX environment depending
+# on the flags and args passed. This script will always compile the code, and run the unit tests if a change was
+# made to something other than a .rst file.
+#
+# Script usage:
+# buildscript <path-to-workspace> <cmake-preset-name> [options]
+#
+# Example command to run a PR build on ubuntu:
+# buildscript /jenkins/workspace_dir/ linux-ci --enable-unittests --enable-coverage --enable-docs
+#
+# Expected args:
+#   1. WORKSPACE: path to the workspace/source code that this should run inside
+#   2. CMAKE_PRESET: the CMake preset that should be ran to generate the cmake files for this CI job
+#
+# Possible flags:
+#   --enable-systemtests: Runs the system tests from being compiled or ran
+#   --enable-package: Runs a package being produced
+#   --enable-docs: Runs the docs from being built
+#   --enable-dev-docs: Runs the developer docs from being built
+#   --enable-doctests: Runs the documentation tests
+#   --clean-build: Clears the build folder and builds from scratch
+#   --clean-external-projects: Clear the external projects from the build folder
+#   --use-core-dumps: Activate core dumps (Linux only)
+#   --extra-cmake-flags: Extra flags to pass directly to cmake, enclose in "".
+
+# SCRIPT_DIR discovery from https://stackoverflow.com/a/9107028 by https://stackoverflow.com/users/1184238/andrew-norrie
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+WORKSPACE=$1
```

Comment:
Could I ask the reason for using shift over just `$1`, `$2`?

Also, I think it would be wise to do a check to see that there are at least 2 arguments and raise an error if not. That way the script doesn't fail weirdly if the unexpected arguments are passed.

Reply:
The reasoning is that the script is using shift later and it simplifies the flag parsing as it always grabs $1, and then $2 if it's taking extra data after the flag.

I will add a check for argument count, ensuring they also aren't flags passed as a first or second argument.